ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -99,7 +99,12 @@ internal constructor(
   ): ImmutableSet<Feature> {
     // TODO: Add support for polylines similar to mapPins.
     return locationsOfInterest
-      .map { Feature(id = it.id, tag = Feature.Type.LOCATION_OF_INTEREST, geometry = it.geometry) }
+      .map {
+        Feature(
+          Feature.Tag(id = it.id, type = Feature.Type.LOCATION_FEATURE, flag = it.job.hasData()),
```

Comment:
Can we roll the two tag args into the constructor of Feature for convenience?

Reply:
At that point I would wonder why we need the additional class wrapping them in the first place and why we wouldn't just store these as direct properties of `Feature`. Rolling them into the primary ctor would make it awkward to do things like:

```kt
val tag = FeatureTag(...)
val feat = Feature(tag, ...)
```

since instead you'd have to do this:

```kt
val tag = FeatureTag(...)
val feat = Feature(tag.x, tag.y, ...)
```

which would seem to make the `Tag` superfluous.

Maybe bundling them up was a premature optimization? (if we go back on this again this will be the second or third time we've changed our minds about this -- the original impl used this approach, I unrolled the fields into the Feature, then you suggested rebundling them into a Tag...now we're considering unrolling them again....)

tbh all we need probably is one flat data class. there's not a huge benefit to having a wrapping class IMO unless we think we have a need to manipulate tags separately from Features.

Reply:
Tag is still needed as an object to be returned by events; there's no need to fetch the geometry from the map and convert it back to a `Feature`.

Expanding the tag args in Feature constructor was suggested as syntactic sugar to aid readability.

Reply:
Gotcha, sg. I'll add a second ctor for convenience